src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -9,11 +9,17 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Hosting.Internal
 {
     public class HostingApplication : IHttpApplication<HostingApplication.Context>
     {
+#if !NET451
```

Comment:
I initially added 4.5 target, but it was removed in #15956 because of some build changes and also it looked like we won't have end-to-end story on .NET 4.5.
But now it seems we'll have it in AppInsight and we need Activity on NET4.5
Let me return it back.